packages/react/src/components/form/flow.ts

```diff
@@ -27,6 +32,8 @@ export interface AuthenticatingState {
   cancel: Cancel;
   recover: () => void;
   entry: () => void;
+  hasUIState: (state: AuthenticatingUIStatus) => boolean;
+  getUiState: () => State<AuthenticatingUIStatus>;
```

Comment:
On a similar note, maybe consider a name that does not imply this must be a UI state? `getNestedState` or `getChildState` maybe?